lightningd/opening_control.c

```diff
@@ -1202,6 +1203,184 @@ static struct command_result *json_fundchannel_start(struct command *cmd,
 	return command_still_pending(cmd);
 }
 
+static struct channel *stub_chan(struct command *cmd,
+								u64 id,
+								struct node_id nodeid, 
+								struct channel_id cid,
+								struct bitcoin_outpoint funding,
+								struct wireaddr_internal addr){
+	struct peer *peer;
+	peer = new_peer(cmd->ld,
+						0,
+						&nodeid,
+						&addr,
+						false
+						);
+	
+	struct lightningd *ld = cmd->ld;
+	
+
+	struct channel_config our_config;
+	memset(&our_config, 1, sizeof(struct channel_config));
+	our_config.id = 0;
+	
+	u32 feerate;
+	/* FIXME: What should be the appropriate value here? */
+	feerate = 192838; 
+
+	struct bitcoin_signature sig;
+	sig.sighash_type = SIGHASH_ALL;
+
+	struct pubkey pk;
+	struct basepoints basepoints;
+	pubkey_from_der(tal_hexdata(ld->wallet, 
+					type_to_string(tmpctx, struct node_id, &nodeid), 66),
+					33,
+					&pk);
+
+	struct pubkey localFundingPubkey;
+	get_channel_basepoints(ld, 
+						   &nodeid, 
+						   id, 
+						   &basepoints, 
+						   &localFundingPubkey);
+
+	struct channel_info *channel_info = tal(ld->wallet, 
+											struct channel_info);
+	memset(channel_info, 3, sizeof(*channel_info));
+	channel_info->their_config.id = 0;
+	channel_info->theirbase = basepoints;
+	channel_info->remote_fundingkey = pk;
+	channel_info->remote_per_commit = pk;
+	channel_info->old_remote_per_commit = pk;
+
+	u32 blockht = 100;
+	
+	struct short_channel_id *scid = tal(cmd, struct short_channel_id);
+	
+	/*To indicate this is an stub channel we keep it's scid to 1x1x1.*/
+	if(!short_channel_id_from_str("1x1x1", 6, scid)){
+		return NULL;
+	}
+	
+	struct channel *channel; //Channel Shell with Dummy data(mostly)
+
+	channel = new_channel(peer, id,
+			      NULL, /* No shachain yet */
+			      CHANNELD_NORMAL,
+			      LOCAL,
+			      peer->ld->log,
+			      "billboard",
```

Comment:
"restored from static channel backup"?  You can see this in listpeers, so it's nice to have....